meilisearch-schema/src/schema.rs

```diff
@@ -0,0 +1,224 @@
+use std::collections::{HashMap, HashSet};
+
+use serde::{Serialize, Deserialize};
+
+use crate::{FieldsMap, FieldId, SResult, Error, IndexedPos};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Schema {
+    fields_map: FieldsMap,
+
+    identifier: FieldId,
+    ranked: HashSet<FieldId>,
+    displayed: HashSet<FieldId>,
+
+    indexed: Vec<FieldId>,
+    indexed_map: HashMap<FieldId, IndexedPos>,
+
+    must_index_new_fields: bool,
+}
+
+impl Schema {
+
+    pub fn with_identifier<S: Into<String>>(name: S) -> Schema {
+        let mut fields_map = FieldsMap::default();
+        let field_id = fields_map.insert(name.into()).unwrap();
+
+        Schema {
+            fields_map,
+            identifier: field_id,
+            ranked: HashSet::new(),
+            displayed: HashSet::new(),
+            indexed: Vec::new(),
+            indexed_map: HashMap::new(),
+            must_index_new_fields: true,
+        }
+    }
+
+    pub fn identifier(&self) -> String {
+        self.fields_map.get_name(self.identifier).unwrap().to_string()
+    }
+
+    pub fn set_identifier(&mut self, id: String) -> SResult<()> {
+        match self.get_id(id.clone()) {
+            Some(id) => {
+                self.identifier = id;
+                Ok(())
+            },
+            None => Err(Error::FieldNameNotFound(id))
+        }
+    }
+
+    pub fn get_id<S: Into<String>>(&self, name: S) -> Option<FieldId> {
+        self.fields_map.get_id(name)
+    }
```

Comment:
Is it necessary to ask for a String? I think a `&str` would be enough, even for the `FieldsMap::get_id` function.